lib/PuppeteerSharp.Tests/PageTests/SetRequestInterceptionTests.cs

```diff
@@ -0,0 +1,356 @@
+            Page.RequestCreated += async (sender, e) =>
+            {
+                requests.Add(e.Request);
+                await e.Request.ContinueAsync();
+            };
+            var response = await Page.GoToAsync(TestConstants.EmptyPage + "#hash");
+            Assert.Equal(HttpStatusCode.OK, response.Status);
+            Assert.Equal(TestConstants.EmptyPage, response.Url);
+            Assert.Single(requests);
+            Assert.Equal(TestConstants.EmptyPage, requests[0].Url);
+        }
+
+        [Fact]
+        public async Task ShouldWorkWithEncodedServer()
+        {
+            // The requestWillBeSent will report encoded URL, whereas interception will
+            // report URL as-is. @see crbug.com/759388
+            await Page.SetRequestInterceptionAsync(true);
+            Page.RequestCreated += async (sender, e) => await e.Request.ContinueAsync();
+            var response = await Page.GoToAsync(TestConstants.ServerUrl + "/some nonexisting page");
+            Assert.Equal(HttpStatusCode.NotFound, response.Status);
+        }
+
+        [Fact]
+        public async Task ShouldWorkWithBadlyEncodedServer()
+        {
+            await Page.SetRequestInterceptionAsync(true);
+            Server.SetRoute("/malformed?rnd=%911", context => Task.CompletedTask);
+            Page.RequestCreated += async (sender, e) => await e.Request.ContinueAsync();
+            var response = await Page.GoToAsync(TestConstants.ServerUrl + "/malformed?rnd=%911");
+            Assert.Equal(HttpStatusCode.OK, response.Status);
+        }
+
+        [Fact]
+        public async Task ShouldWorkWithEncodedServerNegative2()
+        {
+            // The requestWillBeSent will report URL as-is, whereas interception will
+            // report encoded URL for stylesheet. @see crbug.com/759388
+            await Page.SetRequestInterceptionAsync(true);
+            var requests = new List<Request>();
+            Page.RequestCreated += async (sender, e) =>
+            {
+                requests.Add(e.Request);
+                await e.Request.ContinueAsync();
+            };
+            var response = await Page.GoToAsync($"data:text/html,<link rel=\"stylesheet\" href=\"{TestConstants.ServerUrl}/fonts?helvetica|arial\"/>");
+            Assert.Equal(HttpStatusCode.OK, response.Status);
+            Assert.Equal(2, requests.Count);
+            Assert.Equal(HttpStatusCode.NotFound, requests[1].Response.Status);
+        }
+
+        [Fact(Skip = "weired")]
+        public async Task ShouldNotThrowInvalidInterceptionIdIfTheRequestWasCancelled()
+        {
+            await Page.SetContentAsync("<iframe></iframe>");
+            await Page.SetRequestInterceptionAsync(true);
+            Request request = null;
+            Page.RequestCreated += (sender, e) => request = e.Request;
+
+            var _ = Page.QuerySelectorAsync("iframe").EvaluateFunctionAsync<object>("(frame, url) => { frame.src = url; return null; }", TestConstants.ServerUrl);
```

Comment:
Do we need to declare these variables?

Reply:
it's for the warning since we aren't calling `await`